drivers/firmware/imx/se_ctrl.c

```diff
@@ -2280,6 +2275,8 @@ static int se_if_probe(struct platform_device *pdev)
 		ret = se_sysfs_log();
 		if (ret)
 			pr_warn("Warn: Creating sysfs entry for se_log and  se_rcv_msg_timeout: %d\n", ret);
+	} else {
+		kobject_get(se_kobj);
```

Comment:
@ernestvh Just a heads-up: this isn’t really my area of expertise, so feel free to correct me if I’m off.

From what I can tell `se_sysfs_log()` always tries to create a new `/sys/kernel/se` kobject and its files. If it succeeds, it returns 0 (meaning it was just created). If it fails, it cleans up and returns a negative value.

There’s no success path for "already existed".
So if another probe calls it again, `kobject_create_and_add("se", kernel_kobj)` will likely fail due to the name already being used, and you’ll get an error.

Because of that, in this patch, calling `kobject_get(se_kobj)` when ret == 0 (success) will increase the refcount unnecessarily for the first device, which could cause a leak (unless the last remove does two put()).

Reply:
Good points @ernestvh!
Agreed that only the first probe actually calls `se_sysfs_log()`. The main bug I still see is that if a `sysfs_create_file()` fails, we `put()` but never reset `se_kobj`, so later probes will skip creation and end up with a stale pointer and missing files.

I like the idea of moving the `create`/`get` logic fully into `se_sysfs_log()`, it would make the flow clearer and fix the stale pointer case.

Probably best to hear from the driver owner on which pattern they want.
Thanks!